File: 2003/CS101SamplesAll/CS101SamplesBCL2/CreateAService Manager/How-To Service Manager/frmAbout.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace How_To_Service_Manager
{
    public partial class frmAbout : Form
    {
        public frmAbout()
        {
            InitializeComponent();
        }
        // Note: Because this form is opened by frmMain using the ShowDialog command, we simply set the

	// DialogResult property of cmdOK to OK which causes the form to close when clicked.

        private void frmAbout_Load(object sender, System.EventArgs e)
        {

            try
            {

                // Set this Form's Text + Icon properties by using values from the parent form

                this.Text = "About " + this.Owner.Text;

                this.Icon = this.Owner.Icon;

                // Set this Form's Picture Box's image using the parent's icon 

                // However, we need to convert it to a Bitmap since the Picture Box Control

                // will not accept a raw Icon.

                this.pbIcon.Image = this.Owner.Icon.ToBitmap();

                // Set the labels identitying the Title, Version, and Description by

                // reading Assembly meta-data originally entered in the AssemblyInfo.cs file

                // using the AssemblyInfo class defined in the same file

                AssemblyInfo ainfo = new AssemblyInfo();

                this.lblTitle.Text = ainfo.Title;

                this.lblVersion.Text = string.Format("Version {0}", ainfo.Version);

                this.lblCopyright.Text = ainfo.Copyright;

                this.lblDescription.Text = ainfo.Description;

                this.lblCodebase.Text = ainfo.CodeBase;

            }
            catch (System.Exception exp)
            {

                // This catch will trap any unexpected error.

                MessageBox.Show(exp.Message, this.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop);

            }
        }
        private void cmdOK_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: 2003/VB101SamplesAll/VB101SamplesBCL2/CreateAServiceManager/How-To Service Manager/frmAbout.vb
'Copyright (C) 2002 Microsoft Corporation
'All rights reserved.
'THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER 
'EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF 
'MERCHANTIBILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.

'Requires the Trial or Release version of Visual Studio 2005.

Option Strict On

Public Class frmAbout
    ' Note: Because this form is opened by frmMain using the ShowDialog command, we simply set the
    ' DialogResult property of cmdOK to OK which causes the form to close when clicked.
    Private Sub frmAbout_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Try
            ' Set this Form's Text & Icon properties by using values from the parent form
            Me.Text = "About " & Me.Owner.Text
            Me.Icon = Me.Owner.Icon

            ' Set this Form's Picture Box's image using the parent's icon 
            ' However, we need to convert it to a Bitmap since the Picture Box Control
            ' will not accept a raw Icon.
            Me.pbIcon.Image = Me.Owner.Icon.ToBitmap()

            ' Set the labels identitying the Title, Version, and Description by
            ' reading Assembly meta-data originally entered in the AssemblyInfo.vb file
            ' using the AssemblyInfo class defined in the same file
            Dim ainfo As New AssemblyInfo()

            Me.lblTitle.Text = ainfo.Title
            Me.lblVersion.Text = String.Format("Version {0}", ainfo.Version)
            Me.lblCopyright.Text = ainfo.Copyright
            Me.lblDescription.Text = ainfo.Description
            Me.lblCodebase.Text = ainfo.CodeBase

        Catch exp As System.Exception
            ' This catch will trap any unexpected error.
            MessageBox.Show(exp.Message, Me.Text, MessageBoxButtons.OK, MessageBoxIcon.Stop)

        End Try
    End Sub
End Class